### PR TITLE
Universal SQL: Environment Variable Copy Behaviors

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Description

- Environment variables now handle newlines better (This may need some testing for different platforms)
- Environment variables now update properly when you save a connection; so you do not need to refresh the page

### Checklist

- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Review by Winterhart — COMMENTED** [review]
Is the copy/paste available on `next` on the `example-project` ?

File: packages/plugin-connector/src/data-sources/index.js [hunk 1]
```diff
@@ -84,10 +84,11 @@ const generateSourceEnvironmentVariables = (sourceName, sourceConfig) => {
 
 /**
  * Helper function to load configured datasources
+ * @param {string} [datasourceDir]
  * @returns {Promise<(DatasourceSpec & { environmentVariables: object })[]>}
  */
-export async function getDatasourceOptions() {
-	const datasourceDir = await getSourcesDir();
+export async function getDatasourceOptions(datasourceDir) {
+	datasourceDir = datasourceDir ?? (await getSourcesDir()) ?? undefined;
```
[the remark this answers is not in the document]
Comment (ItsMeBrianD) [reply packages/plugin-connector/src/data-sources/index.js:91]: This is a typescript oddity

**Comment by ItsMeBrianD** [comment]
> Is the copy/paste available on next on the example-project ?

Once merged this will be on `next` for `example-project` and `test-env`, but it won't hit the template until a release; I'm planning to cut a release after I get back this evening (8ish ET)

File: packages/core-components/src/lib/unsorted/ui/Deployment/EnvironmentVarListing.svelte [hunk 2]
```diff
@@ -20,7 +20,7 @@
 			return [
 				a,
 				Object.entries(v.environmentVariables)
-					.map(([k, v]) => `${k}=${v}`)
+					.map(([k, v]) => `${k}="${v.replace(/\\n/g, '\n')}"`)
```
[the remark this answers is not in the document]
Comment (ItsMeBrianD) [reply packages/core-components/src/lib/unsorted/ui/Deployment/EnvironmentVarListing.svelte:23]: I was having issues with the ghseets connector _without_ the newlines (it had \n literals), so it's a little unclear. We'll have to see how cloud behaves and go from there I think

**Pull request merged**